generators/profilegen/src/profile_generator.rb

```diff
@@ -0,0 +1,225 @@
+  end
+
+  #
+  # Method to get renderer
+  #
+  # @param template for which the renderer will be created
+  #
+  def get_renderer(template)
+    renderer = ERB.new(template, 0, '-%>')
+    renderer.result(get_binding)
+  end
+
+  def generate_operation_types(resource_provider, resource_type, resource_type_version)
+    namespace = @dir_metadata[resource_provider]['namespace'] + "::#{get_version(resource_type_version)}"
+    namespace = namespace.gsub(/\./, '_' )
+    obj = Module.const_get(namespace)
+    operations = []
+    obj.constants.select do |const_name|
+      if((obj.const_get(const_name).instance_of?Class))
+        if(const_name.to_s.end_with?('ManagementClient') || const_name.to_s == 'ManagementLinkClient' || const_name.to_s == 'ManagementLockClient' || const_name.to_s == 'FeatureClient' || const_name.to_s == 'PolicyClient' || const_name.to_s == 'SubscriptionClient')
+          @management_client = obj.name + '::' + const_name.to_s
+          @versions_clients_mapper[resource_type_version] = @management_client
+        elsif (const_name.to_s == resource_type || '*' == resource_type)
+          operations << const_name.to_s
+        end
+      end
+    end
+
+    if operations.empty?
+      raise "#{resource_type} operation could not be found for RP: #{resource_provider}:Version: #{resource_type_version}"
+    end
+
+    operations.each do |operation|
+      check_and_add_operation(obj, operation, resource_provider)
+    end
+  end
+
+  def check_and_add_operation(module_obj, operation, resource_provider)
+    operation_body = module_obj.name + '::' + operation
+    operation_name_ruby = get_model_name(operation)
+
+    if(check_for_availability(@operation_types, operation_name_ruby))
+      raise "#{operation_name} operation is appearing twice for the RP: #{resource_provider}."
+    else
+      @operation_types << { 'operation_name': operation, 'operation_name_ruby': operation_name_ruby, 'operation_body': operation_body}
+    end
+  end
+
+  def generate_model_types(resource_provider, resource_type_version)
+    namespace = @dir_metadata[resource_provider]['namespace'] + + "::#{get_version(resource_type_version)}::Models"
+    namespace = namespace.gsub(/\./, '_' )
+    obj = Module.const_get(namespace)
+    obj.constants.select do |const_name|
+        model_name = const_name.to_s
+        model_body = obj.name + '::' + const_name.to_s
+        method_name = get_model_name(const_name.to_s)
+
+        if(check_for_availability(@model_types, model_name))
+          raise "#{model_name} Model is appearing twice for the RP: #{resource_provider}."
+        else
+          @model_types << {'model_name': model_name, 'model_body': model_body, 'method_name': method_name}
+        end
+    end
+  end
+
+  def get_model_name(model_name)
+    model_name.gsub(/([a-z\d])([A-Z])/,'\1_\2').downcase
+  end
+
+  def get_client_file
+    check_and_create_directory
+    file_name =  "#{@output_dir}/#{@profile_name}/profile_client.rb"
```

Comment:
Should we use `File.join` instead to take care of the OS specific cases? 

Same comment for all the places in this file where we have used `/`